src/main/java/com/conveyal/gtfs/model/Route.java

```diff
@@ -171,4 +233,184 @@ public Iterator<Route> iterator() {
+
+        routeNetworks.values().forEach(routeNetwork ->
+            routeNetworksByRouteId
+                .computeIfAbsent(routeNetwork.route_id, id -> new HashSet<>())
+                .add(routeNetwork.network_id)
+        );
+
+        routes.values().forEach(route -> route.route_network_ids = getChildIdsMatchingParentId(routeNetworksByRouteId, route.route_id));
+    }
+
+    /**
+     * Merge route networks into routes when loading into DB.
+     */
+    public static CsvReader getCsvReaderForRoutesWithRouteNetworks(
+        CsvReader routesReader,
+        Map<String, Set<String>> routeNetworksByRouteId
+    ) {
+        List<String> rows = new ArrayList<>();
+        try {
+            while (routesReader.readRecord()) {
+                String routeId = routesReader.get(ROUTE_ID_FIELD);
+                rows.add(createRow(routesReader, getChildIdsMatchingParentId(routeNetworksByRouteId, routeId), CSV_FIELDS));
+            }
+            return (rows.isEmpty())
+                ? routesReader
+                : produceCsvPayload(rows, CSV_HEADER_FOR_MERGE);
+        } catch (Exception e) {
+            LOG.error("Error while merging routes", e);
+            // Any issues, return the original routes reader (minus route networks).
+            return routesReader;
+        }
+    }
+
+    /**
+     * Extract the route networks from file and group by route id. This is to allow for easier CRUD by the DT UI.
+     */
+    public static Map<String, Set<String>> groupRouteNetworkIds(CsvReader csvReader, List<String> errors) {
+        Map<String, Set<String>> routeNetworksGroupedByRouteId = new HashMap<>();
+
+        try {
+            while (csvReader.readRecord()) {
+                if (!hasExpectedNumberOfColumns(csvReader, errors, 2)) {
+                    continue;
+                }
+                String routeNetworkId = csvReader.get(RouteNetwork.NETWORK_ID_FIELD);
+                String routeId = csvReader.get(RouteNetwork.ROUTE_ID_FIELD);
+                routeNetworksGroupedByRouteId.computeIfAbsent(routeId, k -> new HashSet<>()).add(routeNetworkId);
+            }
+            return routeNetworksGroupedByRouteId;
+        } catch (IOException e) {
+            return Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Expand all route network ids into a single row for each route id. This is to conform with the GTFS Fares v2 standard.
+     */
+    public static String packRouteNetworks(List<Route> routes) {
+        StringBuilder csvContent = new StringBuilder(createRow(RouteNetwork.ROUTE_ID_FIELD, RouteNetwork.NETWORK_ID_FIELD));
+        routes
+            .stream()
+            .filter(route -> route.route_network_ids != null)
+            .forEach(route -> {
+                String[] routeNetworkIds = route.route_network_ids.split(SEPARATOR);
+                for (String routeNetworkId : routeNetworkIds) {
+                    csvContent.append(createRow(routeNetworkId, route.route_id));
```

Comment:
Fix the order of fields, it is not consistent with the header above.

Reply:
Fixed in: https://github.com/ibi-group/gtfs-lib/pull/16/commits/82684ac3a10245fd73a993a76471ab408648cb57